src/main/java/com/iexec/common/sgx/SgxDriverMode.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.sgx;
+
+public enum SgxDriverMode {
+    NONE(new String[]{}),
+    LEGACY(new String[]{"isgx"}),
+    NATIVE(new String[]{"sgx_enclave", "sgx_provision"});
```

Comment:
```suggestion
    LEGACY(new String[]{"/dev/isgx"}),
    NATIVE(new String[]{"/dev/sgx_enclave", "/dev/sgx_provision"});
```
I would prefer to see the full device path stored here

Reply:
I'm not sure it would be better. Devices should always be put in `/dev` folder on Linux. 
BTW, we use the name without its path there:
https://github.com/iExecBlockchainComputing/iexec-worker/blob/23a4b347a64d9a23bb8549653e1890d24a5eba6d/src/main/java/com/iexec/worker/sgx/SgxService.java#L110

Reply:
I tend to agree with JB, behavior would be more explicit with suggested proposal.

Reply:
I also agree it would make behavior more explicit but it would make [Worker task](https://github.com/iExecBlockchainComputing/iexec-worker/blob/23a4b347a64d9a23bb8549653e1890d24a5eba6d/src/main/java/com/iexec/worker/sgx/SgxService.java#L110) less intuitive.

Reply:
I also think it's better to put the full device path to make things more explicit.
@mcornaton sorry I didn't understand your last comment.

Reply:
Workers need to know the name of the devices to check SGX is supported. As we know devices are always inside `/dev`, we can simply do the following:
```bash
ls /dev | grep -w {list of devices names}
```
If we now store the full path to the devices in `SgxDriverMode`, this command line will become strange. Of course we could prepare devices names by splitting paths before running the command, but it would not be clean.

Reply:
The other solution would be to use `Path` in `SgxDriverMode` such as:
```java
public enum SgxDriverMode {
    NONE(),
    LEGACY(Path.of("/dev", "isgx")),
    NATIVE(Path.of("/dev", "sgx_enclave"), Path.of("/dev", "sgx_provision"))
}
```

But that would lead to the following in `SgxService` of Worker:
```java
        final Path[] devices = sgxDriverMode.getDevices();
        // The following will find SGX devices
        // related to selected SGX driver mode in `/dev`.
        final String cmd = String.format("/bin/sh -c 'echo $(ls /dev | grep -w %s)'",
                Arrays.stream(devices)
                        .map(devicePath -> "-e \"" + devicePath.getFileName() + "\"")
                        .collect(Collectors.joining(" ")));
```
We still explicitly specify the `/dev` folder...

Reply:
With full path, couldn't the test become something like:
`bash -c "test -e <device1> && test -e <device2>"`
Return value will be 0 if all devices exist, 1 otherwise.

Reply:
Same comment, why can't you test like that : https://github.com/scontain/SH/blob/8c3eecc6eda2c22e682e26c492fc1af7a14dca3a/install_sgx_driver.sh#L932 ?

Reply:
Both of you are right, thanks! That's updated in my last commits :smile: